### PR TITLE
[quazip] Set zlib transitive headers

### PR DESCRIPTION
**Pull request by MartinDelille**
zlib is required here: https://github.com/stachenov/quazip/blob/5bd87270bd9c0dbe99c102e5b442e1fe142166be/quazip/zip.h#L56

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
@jcar87 I kind of lost with all these missing dependencies problems. Any idea what is going wrong ?

**Comment by MartinDelille** [comment]
Hi @czoido ! Just a little ping to now if there could be some update on this missing dependencies ?

**Comment by czoido** [comment]
> Hi @czoido ! Just a little ping to now if there could be some update on this missing dependencies ?

Let me try to launch a job to try to generate the missing dependencies and then relaunch this

**Comment by MartinDelille** [comment]
Hi @czoido ! It seems that it is still failing.

**Comment by czoido** [comment]
> Hi @czoido ! It seems that it is still failing.

Hi @MartinDelille,

Thanks for the ping. I changed zlib to use a version range, I think that was the source of the problem with the missing binaries and also added `transitive_headers=True` for qt. Let's wait for ci 🤞

File: recipes/quazip/all/conanfile.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("qt/5.15.9")
-        self.requires("zlib/1.2.13")
+        self.requires("qt/5.15.9", transitive_headers=True)
```
Comment (czoido) [inline recipes/quazip/all/conanfile.py:51]:
```suggestion
        self.requires("qt/5.15.9", transitive_headers=True, transitive_libs=True)
```

**Review by franramirez688 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Comment by czoido** [comment]
@MartinDelille 🥳
Sorry it took so long, we had some problems in the past few weeks due to the high pressure on the ci of the system

**Comment by MartinDelille** [comment]
No worries! Thank you for helping me getting this merged! 🎉

**Pull request merged**